### ISSUE TITLE
List Topic (msg_topic) editing does not update subscribers topic preferences 

### ISSUE THREAD

**Issue by tmclaren** [sympa-community/sympa#1400]
Version
-------
6.2.68

Installation method
-------------------
rpm

Expected behavior
-----------------
If a list owner updates/edits a topic name, the new name should propagate to the subscribers preferences.

Actual behavior
---------------
list owner defines 'topic1' and changes/edits the topic name to 'topic-a'.  This new name does not propagate to the list subscribers that added 'topic1' to their preferences, topic1 persists in subscriber preferences (subscriber_table > topics_subscriber)

Steps to reproduce
------------------
1. Create a new list, 
.2 define a topic, 'topic1', 
3. add subscribers, 
4. change subscriber preferences to select 'topic1', 
5. save.  
6. list owner changes topic1 name to topic-a.
7. view subscribers page and the old topic1 name persists for subscribers that selected topic1.  Sympa did not update the subscriber_table to reflect the new name.
 
Additional information
----------------------
A work-around requires each subscriber to reset their preference to the new topic name, topic-a.

**Comment by tmclaren** [comment]
For this case, I was able to update the db using:

MariaDB [sympa]>  UPDATE subscriber_table SET topics_subscriber = REPLACE(topics_subscriber, 'SUMARY', 'SUMMARY'); 
Query OK, 107 rows affected (2.531 sec)
Rows matched: 647403  Changed: 107  Warnings: 0

NOTE:  This query/updated the type-o across all list topics vs a single list.

**Comment by ikedas** [comment]
Just a comment:

IMHO once a topic is added, its name should not be changed, i.e. another topic should be added to add topic with another name.

**Comment by racke** [comment]
I agree with @tmclaren. You really want to change a topic when it becomes offensive for some reason.

**Comment by ikedas** [comment]
Sorry. My comment above is not a direct fix to this bug. What I meant was that:
  - As the name is the unique identifier of the topic, if the name changes, it is hard to determine which topic has been edited (imagine changing the name of several topics at once).
  - Thus, the name change operation would be better to be replaced by the topic add/delete operation.

For example, please visit the Users - Owners in list Admin page.  As email is unique identifier, changing email of each owner is impossible, but addition/deletion of owners are possible.

Anyway, this is not a direct fix for the bug discussed on this issue page.

**Comment by tmclaren** [comment]
It's unfortunate the data model doesn't support single or multiple name
changes; seems like a design flaw.    The problem with the add/delete
approach is the subscribers would have to take action to select the new
topic.

W.R.T. owner email addresses, are you saying if an owner changes their
email address in "My Preferences" that change will not propagate through
the db and update their subscriber, moderator, or owner roles?

**Comment by ikedas** [comment]
> W.R.T. owner email addresses, are you saying if an owner changes their email address in "My Preferences" that change will not propagate through the db and update their subscriber, moderator, or owner roles?

No.  It's not the case.

**Comment by bboyle262** [comment]
Hello, just wanted to follow-up on this issue and see if there were any plans in future sympa releases to allow for name changes of topics.

Thanks

**Comment by ikedas** [comment]
No one has proposed a specific way to achieve direct modification of topic identifier. Therefore, there are currently no plans to do so in future Sympa releases.

**Comment by ikedas** [comment]
To begin with, there is no need to assign a meaningful string to the "name" of the message topic. The meaning of the topic is explained in the title.  We may use an auto-generated unique id for the topic "name" and hide it from the end-users.
